mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -648,4 +662,92 @@ mod tests {
 
         assert_eq!(expected_user_agent, user_agent_header_value);
     }
+
+    #[test]
+    #[ignore]
+    fn test_create_client_with_valid_credentials() {
+        let expected_user_agent = "mountpoint-s3-client";
+        let profile_name_override = String::from("VALID_CREDENTIALS");
+        let config = S3ClientConfig {
+            profile_name_override: Some(profile_name_override),
+            ..Default::default()
+        };
+
+        let client = S3CrtClient::new("eu-west-1", config).expect("Create test client");
+
+        let mut message = client
+            .new_request_template("GET", TEST_BUCKET_NAME)
+            .expect("new request template expected");
+
+        let headers = message.inner.get_headers().expect("Expected a block of HTTP headers");
+
+        let user_agent_header = headers.get("User-Agent").expect("user agent header should be set");
+
+        let user_agent_header_value = user_agent_header.value();
+
+        assert_eq!(expected_user_agent, user_agent_header_value);
+    }
+
+    #[test]
+    fn test_create_client_with_non_existent_credentials() {
+        delete_credentials();
+
+        let profile_name_override = String::from("NONEXISTENT_CREDENTIALS");
+        let config = S3ClientConfig {
+            profile_name_override: Some(profile_name_override),
+            ..Default::default()
+        };
+        let error = S3CrtClient::new("eu-west-1", config).unwrap_err();
+        let expected_error = NewClientError::ProviderFailure(CrtErr::from(AWS_ERROR_FILE_INVALID_PATH));
+
+        assert_eq!(error, expected_error);
+    }
+
+    #[test]
+    fn test_create_client_with_invalid_credentials() {
```

Comment:
Making request is the only way for detecting  invalid credentials. However, if we move this to integration test, we have to make `s3_crt_client module` and `makeSimpleHttp` request public. So I think it would be fine to keep this type of test here.

Reply:
I don't really understand this test anyway -- it doesn't check that the error is right. Seems like we could just move the test to `tests/` where we have a real S3 bucket to test against, so we can verify that the failure is due to credentials rather than the bucket not existing or something.